README.md

```diff
@@ -2,17 +2,161 @@
+```go
+	gremlinServer := "ws://server1:8182/gremlin"
+	maxPoolCapacity := 10
+	maxRetries := 3
+	verboseLogging := true
+	pingInterval := 5
+
+	myStack, err := gremlin.NewGremlinStackSimple(gremlinServer, maxPoolCapacity, maxRetries, verboseLogging, pingInterval)
+	if err != nil {
+		// handle error here
+	}
+```
+
+Note:
+The arguments for NewGremlinStack perform the following functions:
+
+* maxPoolCapacity sets the number connections to keep in the pool
+
+* maxRetries sets the number of retries to use on a connection before trying another
+
+* verboseLogging denotes the verbosity of logs received from your gremlin server
+
+* pingInterval denotes how often (in seconds) the gremlin pool should refresh itself
+
+### Querying the database
+
+To query the database, using your stack, call `ExecQueryF()`, which requires a Context.context object:
+```go
+
+	ctx, _ := context.WithCancel(context.Background())
+	query := g.V("vertexID")
+	response, err := myStack.ExecQueryF(ctx, query)
+	if err != nil {
+		// handle error here
+	}
+```
+
+Optionally, you may parameterize the query and pass arguments into `ExecQueryF()`:
+```go
+	var args []interface{
+		"vertexID",
+		"10"
+	}
+	query := "g.V().hasLabel("%s").limit(%d)"
+	response, err := myStack.ExecQueryF(ctx, query)
+	if err != nil {
+		// handle error here
+	}
+
+```
+The gremlin client will handle the interpolation of the query, and escaping any characters that might be necessary.
+
+### Reading the response
+
+`ExecQueryF()` will return the string response returned by your Gremlin instance, which may look something like this:
+```json
+	[
+	  {
+	    "@type": "g:Vertex",
+	    "@value": {
+	      "id": "test-id",
+	      "label": "label",
+	      "properties": {
+	        "health": [
+	          {
+	            "@type": "g:VertexProperty",
+	            "@value": {
+	              "id": {
+	                "@type": "Type",
+	                "@value": 1
+	              },
+	              "value": "1",
+	              "label": "health"
+	            }
+	          }
+	        ]
+	      }
+	    }
+	  }
+	]
+```
+
+This library provides a few helpers to serialize this response into Go structs. There are specific helpers for serializing a list of vertexes and for a list of edges, and there is additionally a generic vertex that serializes the response into a struct with a type & a value interface{}.
+
+`SerializeVertexes()` would turn the response above into the following, which is considerably easier to manipulate:
+```go
+	Vertex{
+		Type: "g:Vertex",
+		Value: VertexValue{
+			ID:    "test-id",
+			Label: "label",
+			Properties: map[string][]VertexProperty{
+				"health": []VertexProperty{
+					VertexProperty{
+						Type: "g:VertexProperty",
+						Value: VertexPropertyValue{
+							ID: GenericValue{
+								Type:  "Type",
+								Value: 1,
+							},
+							Value: "1",
+							Label: "health",
+						},
```

Comment:
I like the idea, but the implementation might need a little tinkering, because it may cause a breaking change in the data-platform-service.

Also, we'll need to keep that ID field (or at least should, because it's a unique ID for the property in the database, I believe)

Reply:
now that I think about it, it may be better to put this off until after the demo on friday though

Reply:
I'll make a separate ticket to do this